File: lib/std/java/java/nio/file/InputFile.java
package java.nio.file;

public class InputFile {

    private String fileName;

    InputFile(String filename) {
        this.fileName = filename;
        open(filename);
    }

    public native void open(String filename);
    public native void close();
    public native String readLine();
    public native String readFile();
}

File: lib/mrjvm/native/file/input_file.rb
class InputFile
  attr_reader :filename

  def initialize(filename)
    @filename = filename
    @file = File.open(@filename, 'r')
  end

  def read_line
    @file.gets
  end

  def read_file
    @file.read
  end

  def close
    @file.close
  end
end
